### PR TITLE
Feature/stripe tables

### PR DESCRIPTION
**Pull request by elijah-rou**
Stripe dbt Tables v1

### CODE REVIEW

**Comment by elijah-rou** [comment]
To Do:

- Customer Stats Table (with Subs)
- Daily Stats Table (full date range with Subs)

File: models/schema.yml [hunk 1]
```diff
@@ -16,3 +28,172 @@ models:
...
+        description: "The unique id of the subscription purchase (if applicable)"
+      - name: subscription_billing
+        description: "The billing used for the subscription purchase"
+      - name: subscription_start_date
+        description: "The start date of the subscription purchase"
+      - name: subscription_end_date
+        description: "The end date of the subscription purchase"
+      - name: plan_id
+        description: "The plan associated with the purchased subscription"
+      - name: plan_is_active
```
Comment (milo157) [inline models/schema.yml]: Should be is_plan_active

File: models/tmp/stg_stripe__customers.sql [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+select
+    id as customer_id,
+    {{ dbt_date.from_unixtimestamp('created') }} as created_at,
+    description as customer_description,
+    email as customer_email,
+    delinquent as is_delinquent,
+    currency as customer_currency,
+    default_card as default_card_id,
+    shipping as shipping_name,
```
Comment (milo157) [inline models/tmp/stg_stripe__customers.sql:21]: is it shipping or billing address?

File: models/stripe__customers.sql [hunk 3]
```diff
@@ -1,33 +1,197 @@
...
-commit_comment_reactions_customer as (
+transactions_by_customer as (
     select
-        id as customer_id,
-        url,
-        type,
-        login as customer_name
-    from {{ var('commit_comment_reactions_customer') }}
+        customer_id,
+        sum(case when type in ('charge', 'payment')
```
Comment (milo157) [inline models/stripe__customers.sql]: fix the linting where. Usually easiest to read a case like:
case
       when type in ('charge', 'payment')  then type else 0
end as total

File: models/stripe__customers.sql [hunk 4]
```diff
@@ -1,33 +1,197 @@
...
+        coalesce(transactions_by_customer.refund_count_this_month, 0) as refund_count_this_month,
+        transactions_by_customer.first_sale_date,
+        transactions_by_customer.most_recent_sale_date,
+        coalesce(failed_charges_by_customer.total_failed_charge_count, 0) as total_failed_charge_count,
+        coalesce(failed_charges_by_customer.total_failed_charge_amount/100, 0) as total_failed_charge_amount,
+        coalesce(failed_charges_by_customer.failed_charge_count_this_month, 0) as failed_charge_count_this_month,
+        coalesce(failed_charges_by_customer.failed_charge_amount_this_month/100, 0) as failed_charge_amount_this_month,
+        customers.customer_currency,
+        customers.default_card_id,
+        customers.shipping_name,
```
Comment (milo157) [inline models/stripe__customers.sql:223]: shipping or billing?

**Review by milo157 — COMMENTED** [review]
A few comments but have a few questions we should discuss after standup:

- Conventions of tmp files
- Difference between balance transactions/charges

File: models/stripe__customers.sql [hunk 5]
```diff
@@ -1,33 +1,199 @@
...
+        coalesce(transactions_by_customer.fees_this_month/100.0, 0) as fees_this_month,
+        coalesce(transactions_by_customer.net_transaction_amount_this_month/100.0, 0) as net_transaction_amount_this_month,
+        coalesce(transactions_by_customer.sales_count_this_month, 0) as sales_count_this_month,
+        coalesce(transactions_by_customer.refund_count_this_month, 0) as refund_count_this_month,
+        transactions_by_customer.first_sale_date,
+        transactions_by_customer.most_recent_sale_date,
+        0 as total_failed_charge_count,
+        0 as total_failed_charge_amount,
+        0 as failed_charge_count_this_month,
+        0 as failed_charge_amount_this_month,
```
Comment (jonoirwinrsa) [inline models/stripe__customers.sql:177]: just checking - should these be 0's or was that a kind of placeholder?

File: models/stripe__invoice_line_items.sql [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+        *
+    from {{ ref('stg_stripe__plans') }}
+),
+
+line_items_summary as (
+    select 
+        -- Invoices
+        invoices.invoice_id,
+        invoices.invoice_number,
+        invoices.created_at as invoice_created_at,
```
Comment (jonoirwinrsa) [inline models/stripe__invoice_line_items.sql:44]: should these created_at's not have _timestamp

File: packages.yml [hunk 7]
```diff
@@ -1,5 +1,5 @@
 packages:
-  - package: dbt-labs/dbt_utils
```
Comment (jonoirwinrsa) [inline packages.yml:2]: 🤔 looks like you are using dbt_utils above? should this have been removed?

**Review by jonoirwinrsa — APPROVED** [review]
it looks good to me but there is a lot going on. Perhaps this is worthy of a pair-review where you walk us through the code?

File: packages.yml (hunk 7 above)
Comment (elijah-rou) [reply packages.yml:2]: it's part of dbt_date, says on the dbt_date repo that having both may cause conflicts

File: models/stripe__invoice_line_items.sql (hunk 6 above)
Comment (elijah-rou) [reply models/stripe__invoice_line_items.sql:44]: nope cause I convert everything from timestamp to date time

File: models/stripe__customers.sql (hunk 5 above)
Comment (elijah-rou) [reply models/stripe__customers.sql:177]: they should be zero

**Review by milo157 — APPROVED** [review]
LGTM except the few remaining small comments. Before making public we should test with our stripe account.

File: models/stripe__customers.sql (hunk 3 above)
Comment (elijah-rou) [reply models/stripe__customers.sql]: Just changed it, most of the case statements are beefy though so I've adopted this format to make it readable:
```
case when
        condition_1
        and condition_2
    then x
    else y end
) as column_name
```

**Comment by elijah-rou** [comment]
Sent a message to Stripe support for some clarification on their MRR calculation. It's currently very difficult to determine if my calculation is accurate or not as everything is in pounds and I don't know the exchange rate they used. Proceeding with Klaviyo for now instead.

**Pull request merged**